src/nvim/fileio.c

```diff
@@ -5181,7 +5181,10 @@ static void vim_mktempdir(void)
 
     // "/tmp/" exists, now try to create "/tmp/nvim.<user>/".
     add_pathsep(tmp);
-    xstrlcat(tmp, "nvim.", sizeof(tmp));
+
+    const char *appname = get_appname();
+    xstrlcat(tmp, appname, sizeof(tmp));
+    xstrlcat(tmp, ".", sizeof(tmp));
```

Comment:
@glacambre do you remember any motivation for this change to `vim_mktempdir` specifically? The way that `vim_mktempdir` is intended to work, is that it creates a single top-level `nvim.<user>` directory in `/tmp`, and each Nvim instance creates a `nvim.<user>/subdir` where it stores its actual temp files.

So I'm thinking to revert this change, which should fix https://github.com/neovim/neovim/issues/30256

Reply:
I think when I implemented $NVIM_APPNAME, the usecase I had in mind was applications that do something very different from each other (e.g. mail clients embedding nvim, chat clients embedding nvim and so on). From that point of view, it made little sense to desire a way to connect to all neovims running on the same machine, since it's very unlikely that you were going to want to perform the same operations on all of them (as opposed to, say, multiple GUIs running neovim instances dedicated to editing code). This is also the reason I wanted to prevent users from using relative paths in $NVIM_APPNAME with #23211 , I thought it needed to be a real "name" perhaps even the one of the binary that was embedding nvim.

The community has proved countless times that the usecase I had in mind was very far from what it wants to do with $NVIM_APPNAME, so I think we could indeed revert this part :).

Reply:
Thanks, see https://github.com/neovim/neovim/pull/30310

> usecase I had in mind was applications that do something very different from each other (e.g. mail clients embedding nvim, chat clients embedding nvim and so on). From that point of view, it made little sense to desire a way to connect to all neovims running on the same machine

iI think we can still support that use-case even if all the nvim tempdirs are under one `/tmp/nvim.user/` root. The nvim servernames stored in each `/tmp/nvim.user/xxx/<appname>.n.m` use the appname. And eventually, for a given `$NVIM_APPNAME` we could keep a list of "peers" in the app-specific data dir.